vol_object_test.c

```diff
@@ -7038,6 +7182,14 @@ object_visit_callback(hid_t o_id, const char *name, const H5O_info2_t *object_in
 
     UNUSED(o_id);
 
+    if (!HDstrcmp(name, OBJECT_VISIT_TEST_GROUP_NAME_PARENT) ||
+        !HDstrcmp(name, OBJECT_VISIT_TEST_GROUP_NAME_PARENT "/" OBJECT_VISIT_TEST_GROUP_NAME_CHILD) ||
+        !HDstrcmp(name, OBJECT_VISIT_TEST_GROUP_NAME_PARENT "/" OBJECT_VISIT_TEST_GROUP_NAME_CHILD
+                                                            "/" OBJECT_VISIT_TEST_GROUP_NAME_GRANDCHILD)) {
+        (*i)--;
+        goto done;
+    }
```

Comment:
I'd suggest rearranging the group hierarchy or even putting the nested groups in the separate new file created so that you don't have to worry about decrementing the counter here in other unrelated parts of the H5Ovisit test. That way the two different parts aren't dependent on each other and won't get out of sync and require more `strcmp` logic here if more nested groups are added.